### ISSUE TITLE
Kickstart the short code process for 2FA

### ISSUE THREAD

**Issue by jimleroyer** [cds-snc/notification-planning-core#182]
# Description

As a product manager,
I want to increase the sending rates for SMS within GCNotify,
And its reliability,
So that I can send faster, from a trusted short code number.

## WHY are we building?

* Faster throughput
* More reliable
* More trust from users

## WHAT are we building?

Making a request for short code. That does not cover integration, migration, etc...

## VALUE created by our solution

* Experience with short codes.
* Faster throughput.
* More trust.

# Acceptance Criteria

- [X] Sync with management / product on short code status.
- [ ] Open an AWS ticket that request the short code. This should start AWS collaboration officially.
- [ ] Bring a classification for generic usages for short codes within GCNotify. Done on a good will gesture toward AWS and telecoms. We know we want to start with 2FA.
- [ ] Update the ADR on short code.
- [ ] Keep this ticket alive until we get the actual short code.

# QA Steps

- [ ] Review the ADR updates.
- [ ] Review the usage classifications.

**Comment by jimleroyer** [comment]
* Provided short code updates to Josh yesterday.
* I will open the AWS ticket to request a short code today.

**Comment by jimleroyer** [comment]
Jimmy to create the AWS ticket today.

**Comment by yaelberger-commits** [comment]
Wendy connected us with a contact at CRA to discuss their short code. Yael will respond today

**Comment by sastels** [comment]
Will get the ticket in soon

**Comment by sastels** [comment]
Ticket almost ready to send

**Comment by jimleroyer** [comment]
I sent the AWS ticket some minutes ago. I will entertain the AWS replies for today. Our AWS representative is on vacation and have a back up, it might be more difficult for the ticket to get in touch with our SMS contact in AWS.

**Comment by jimleroyer** [comment]
AWS got back to us. Navpreet is back from vacation and looking into our request. Jimmy to reply back to ask for further clarifications on next steps.

**Comment by jimleroyer** [comment]
I sent a reply this morning on how to reach Navpreet for updates. Waiting on AWS.

**Comment by sastels** [comment]
Will check with Pierre Antoine and Navpreet.

**Comment by jimleroyer** [comment]
Pierre-Antoine contacted me saying that Navpreet needs our request application document. On my end, I am thinking 2 different applications, one for 2FA authentication and one for a generic use. I started to look closely into the documentation txt.ca to dig further into the constraints. Also, thinking of communicating directly with the txt.ca folks as they provide a phone number and email.

**Comment by jimleroyer** [comment]
I am doing the 2FA application request for now; to follow up with the generic one.

**Comment by jimleroyer** [comment]
My goal is to complete 2FA app request for today.

**Comment by jimleroyer** [comment]
The request for the 2FA is written, except for the volume data that I need to query. Today, I will do the generic application one.

**Comment by ben851** [comment]
Request document is ready to go. Jimmy to speak with Pierre Antoinne and will see if a meeting can be scheduled

**Comment by jimleroyer** [comment]
Blazer query that contains estimated number of 2FA SMS max per minute for given days. This contains a list of service IDs that has 2FA authentication (email or SMS -- but the query itself restricts to SMS):
http://localhost:1338/queries/136-support-throughput-send-rate-sms-2fa?start_time=2023-07-01T00%3A00%3A00Z&end_time=2023-10-02T23%3A59%3A59Z

Results:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMjhEQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--24a5f24e8e8187695850346f7977cc877183bc5a/image.png)

The 2A short code request document:
https://docs.google.com/document/d/1h65ekLt_CLyQs4uqvNqTch0071afC0iNYvUukU1-s0I/edit

**Comment by jimleroyer** [comment]
I spoke with our AWS TAM about the provided document and I add one more estimate on messages volume for a monthly basis.

**Comment by jimleroyer** [comment]
I provided a more recent version of the document yesterday to our AWS TAM:
https://docs.google.com/document/d/1h65ekLt_CLyQs4uqvNqTch0071afC0iNYvUukU1-s0I/edit

I also had a new a new Blazer query to get monthly volume of the 2FA related sent SMS:
http://localhost:1338/queries/138-support-total-send-rate-sms-2fa?start_time=2023-07-01T00%3A00%3A00Z&end_time=2023-10-10T23%3A59%3A59Z

**Comment by sastels** [comment]
waiting for AWS feedback

**Comment by jimleroyer** [comment]
I contacted our AWS TAM 2 days ago. He said he contacted the short code team and they didn't reply back.

**Comment by ben851** [comment]
No update from AWS. Jimmy to follow up...

**Comment by jimleroyer** [comment]
> Navpreet a repris le dossier pour discussion a l’interne et escalade selon les dernieres informations (avant-hier)

**Comment by jimleroyer** [comment]
From PA:
> J’ai vu passer un message de navpreet en date du 25 octobre, en discussion avec les aggregateurs.
> pour voir si il y a un bloquant dans la demande

**Comment by sastels** [comment]
Might contact txt.ca folks directly

**Comment by jimleroyer** [comment]
I will have an AWS cadence meeting and I will raise the short code question in there. I have meeting with Yael on Wednesday about meeting with the txt.ca folks.

**Comment by ben851** [comment]
Jimmy has met with Yael to plan the next steps.

**Comment by jimleroyer** [comment]
We received an email yesterday from AWS on the ticket that requests us to fill in another request. I am confused about this and I will ask about it.

**Comment by jimleroyer** [comment]
I replied back to the tickets from AWS that ask us to refill the initial request. They said they will look into it (with another canned response).

**Comment by sastels** [comment]
reviewing and providing the data they now require

**Comment by ben851** [comment]
Max sent a mean message to AWS TAM's a meeting will happen soon to discuss requirements

**Comment by jimleroyer** [comment]
New document is almost filled, we just need the SMS replies to be set:
https://docs.google.com/document/d/1p_38ZHvfh0pK3Zy3DI7-Ck20mMQwfoAs/edit

**Comment by ben851** [comment]
Received a reply from AWS yesterday, and Jimmy met with support in AWS cadence meeting.  We need to pay $$ to proceed. Jimmy is seeking the approval form Ioana for the short code costs.

**Comment by ben851** [comment]
Leanne approved the costing. AWS has been notified that we are approved, and the request has been sent to the aggregators.

Lots of intermediaries between us and the telecoms...

**Comment by ben851** [comment]
Waiting on Telecoms/Aggregators.

**Comment by jimleroyer** [comment]
I had conversations yesterday with Melissa and Ioana to find contact's email for telecoms, for our second approach to speak with said telecoms.

**Comment by jimleroyer** [comment]
We had a response from the aggregators. They want to know about the opt-in process. I forwarded the questions to the team internally. Policy looked into it and is currently determining a proper response to this.

**Comment by ben851** [comment]
AWS sent a message saying that the downstream providers submitted the request to the CTA, it will be reviewed January 15.

**Comment by jimleroyer** [comment]
Meeting next Monday with the txt.ca team.

**Comment by jimleroyer** [comment]
We got the short code yesterday by end of day! 🕺💃🎶

**Comment by jimleroyer** [comment]
Jimmy to update the existing ADR on short code acquisition and kick off a new one on integration.

**Comment by jimleroyer** [comment]
We met with the CWTA representative yesterday. It went pretty well as she understands our cases and think we could get an exception, given we are gov and depending our guidelines and safety measures. Next steps:

1- Send to the CWTA representative the name of our AWS TAM.
2- Send to the CWTA representative previous reasoning for short code  denials from the CWTA.

**Comment by jimleroyer** [comment]
Considering this one done as we did acquire a short code! and I opened another ticket to try to migrate our new short code into a generic one #278

**Comment by ben851** [comment]
Keeping this in review to track updating the ADR

**Comment by jimleroyer** [comment]
Updates on the short code ADR ready for review:
https://github.com/cds-snc/notification-adr/pull/55/files

**Comment by ben851** [comment]
@ben851 to review this morning

**Comment by jimleroyer** [comment]
The ADR changes for the short code has been reviewed by Yael and Ben. This has now been merged. It's ready to be moved to done by the kanban master of the universe.
